plugins/SegmentEditor/Model.php

```diff
@@ -129,6 +129,87 @@ public function getSegmentByDefinition($definition)
         return $segment;
     }
 
+    /**
+     * Gets a list of segments that have been deleted in the last week and therefore may have orphaned archives.
+     * @param Date $date Segments deleted on or after this date will be returned.
+     * @return array of segments. The segments are only populated with the fields needed for archive invalidation
+     * (e.g. definition, enable_only_idsite).
+     * @throws \Exception
+     */
+    public function getSegmentsDeletedSince(Date $date)
+    {
+        $dateStr = $date->getDatetime();
```

Comment:
@katebutler not 100% sure why but for my MySQL version it works only when using `getTimestamp()` instead of `getDatetime()`. Makes kind of sense since the data is stored as timestamp but not sure how it works somewhere else.

Can you test to use a `getTimestamp()` and see what it does for you if tests still pass? Then this seems to be the safer option

Reply:
As an example `$date->getTimestamp()` is 1567934627 for 2019-09-08. On my localhost when I use that value in the query it matches any record with `ts_last_edit` not null, even if I set it right back as close to epoch as mysql will let me go (1970-01-01 12:01). This causes the relevant test to fail as the code picks up records that have been deleted more than a week ago.
I am on quite an old version of mysql (5.7.27).

Reply:
All good! The date in my virtual machine was outdated :( works 👍